### PR TITLE
Reorganizing into several source files.

### PR DESCRIPTION
**Pull request by dcoombs**
Have a look at this and see what you think.  I know you had some hesitation about using macros to reduce duplicate code, but I do think this is an improvement and not so hard to follow, hopefully you'll agree.  If not, happy to discuss alternatives. :)

Split pkcs11.c into a separate C file for each PHP class.  An internal-only
header, pkcs11int.h, defines the objects, and preprocessor macros to define
the ctor, dtor, and a registration function for each class.  Also added the
CK_RV to the exception message when pkcs11_error() is called, and added a
general_error() function for cases where there is no CK_RV.

### CODE REVIEW

**Comment by gamringer** [comment]
I merged because it seemed good, without actually testing :(

I'm getting this error when trying to run:

> symbol lookup error: modules/pkcs11.so: undefined symbol: register_pkcs11

**Comment by dcoombs** [comment]
Did you re autom4te and configure again and make clean and make?

**Comment by gamringer** [comment]
Well, look at me looking dumb...
